### DISCUSSION TITLE
Github actions IP ranges?

### DISCUSSION THREAD

**Discussion by georgyfarniev** [community/community#26442] [Actions]
<p>Hello, I need to know subnets of github actions, because my staging server blocks access by default and need to whitelist specific IP/Subnet, it’s necessary to run e2e tests in Github Actions against my staging server. Please help.</p>

**Comment by eddiechenms** [comment, marked as answer]
<p>GitHub Hosted Windows or Ububtu runner use the same IP address arranges as Azure Data Center. You can following our documentation here to get the IP address: <a href="https://help.github.com/en/actions/automating-your-workflow-with-github-actions/virtual-environments-for-github-hosted-runners#ip-addresses-of-github-hosted-runners" rel="nofollow noopener">IP addresses of GitHub-hosted runners</a>.</p>
<p>By the way, <a href="https://github.blog/2019-11-05-self-hosted-runners-for-github-actions-is-now-in-beta/" rel="nofollow noopener">Self-hosted runners</a> is now in beta, you can also try with it if you want to control the runner by yourself and whitelist the runners more eaisly.</p>

**Comment by aerogus** [comment]
<p>Hi,</p>
<p>I’m discovering GitHub Action. It seems to be great. but…</p>
<p>are we sure the ip ranges provided by Microsoft is always up to date ?</p>
<ul>
<li>I’ve just downloaded from MS the current official file named <em>ServiceTags_Public_20200112.json</em>
</li>
<li>I have trigged, with a GitHub-hosted runner, a GitHub Action to build my project and deploy to my Debian server (with strict firewall rules). It’s a simple <em>rsync</em> command</li>
<li>It failed (connection timeout)</li>
<li>in <em>/var/log/messages</em> I see that an ip (ex: <em>40.70.58.112</em> but it is variable) has tried to access my server (at the very same time of the action so it is an ip from GitHub Action). But this ip <strong>does not belong</strong> to the json file !</li>
<li>When I disable my firewall, the connection is done (I have others problems but it’s another story :wink:
</li>
</ul>
<p>Thx for your help</p>

**Comment by juusujanar** [comment]
<p>I just checked the latest IP ranges, same filename as yours. The IP is there, as it belongs to subnet 40.70.0.0/18.</p>

**Comment by aerogus** [comment]
<p>Thanks for taking the time about my issue.</p>
<p>I had done a mistake in my iptables rules where the order of the lines is important.</p>
<p>I was logging all rejected packets in the end of my main firewall script, and added allow rules for the azure ip only after. So theses azure rules were never reached.</p>
<p>Just an iptables noobs problem :slight_smile:</p>

**Comment by dandv** [comment]
<p>This thread is a duplicate.</p>
<p>The current thread is <a href="https://github.community/t/whitelist-github-workflow/16077">https://github.community/t5/GitHub-Actions/Whitelist-Github-Workflow/td-p/15916</a></p>
